spec/classes/jenkins_config_spec.rb

```diff
@@ -19,6 +19,11 @@
       let(:params) { { :config_hash => { 'AJP_PORT' => { 'value' => '1234' } } }}
       it { should contain_jenkins__sysconfig('AJP_PORT').with_value('1234') }
     end
+
+    context 'use config template' do
+      let(:params) { { :jenkins_config_template => 'jenkins/_config.xml.erb' } }
+      it { should contain_class('jenkins::config::global::jenkins_config')}
```

Comment:
Instead of testing for the class, it would be better to test for the `file` resource managing `config.xml`.  One of the most important things to test is resource ordering.